ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/DefaultSchemaInjector.java

```diff
@@ -122,6 +123,7 @@ public DefaultSchemaInjector(final TopicSchemaSupplier schemaSupplier) {
         props.getKafkaTopic(),
         props.getKeySchemaId(),
         keyFormat,
+        SerdeFeatures.of(SerdeFeature.UNWRAP_SINGLES),
```

Comment:
First off, you should only be setting this flag if the format supports unwrapping!  Or more correctly, if the format supports both unwrapping and wrapping.

As to hard coding it... humm, yes, so far it's been hard-coded in one place: `SerdeFeaturesFactory`.  

We could do something like:

```suggestion
        SerdeFeaturesFactory.buildInternal(FormatFactory.of(keyFormat),
```

But, to be honest, this is only a temporary thing.  At the moment, we only support unwrapped keys. Once we support something else we'll wire it up and this hardcoding will go.

Reply:
Good call on only setting the feature if it's supported.

I worry the naming of `buildInternal()` will be confusing here since it's not clear what internal topics have to do with schema registry. Is it worth introducing another alias for the method such as `setUnwrappingIfSupported()` or would that only make things worse?